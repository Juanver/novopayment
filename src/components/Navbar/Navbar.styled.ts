import styled from "styled-components";
import { Palette } from "../../assets/palette/Palette";
import { Spacing } from "../../assets/spacing/Spacing";

const ContainerNavbar = styled.div`
  background-color: ${Palette.BrandBlue};
  color: white;
  display: flex;
  justify-content: space-between;
  padding: ${Spacing.Spacing1} ${Spacing.Spacing2};
`;
const ContainerUserName = styled.div`
  display: flex;
  gap: ${Spacing.Spacing3};

  img {
    width: ${Spacing.Spacing2}
  }
`;

export const StyledNavbar = {
  ContainerNavbar,
  ContainerUserName,
};