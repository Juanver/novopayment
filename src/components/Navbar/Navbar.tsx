import React from 'react'
import { StyledNavbar } from './Navbar.styled';
import PrutecImage from '../../assets/TuPruTec.svg'
import ArrowDown from '../../assets/ArrowDown.svg'

export const Navbar: React.FC = () => {
  return (
    <StyledNavbar.ContainerNavbar>
      <img src={PrutecImage} alt="Tu Prutect logo"/>
      <StyledNavbar.ContainerUserName>
        <p>USER NAME</p>
        <img src={ArrowDown} alt="Arrow down"/>
      </StyledNavbar.ContainerUserName>
    </StyledNavbar.ContainerNavbar>
  )
}
