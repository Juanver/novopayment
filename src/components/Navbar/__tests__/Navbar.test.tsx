import { Navbar } from "../Navbar";
import { render, screen } from "@testing-library/react"

describe('Navbar component', () => {
    it('Should render', async () => {
        const { container } = render(<Navbar />);
        expect(container).toMatchSnapshot();
    });

    it('Should show the user name text', async () => {
        render(<Navbar />);
        const text = screen.queryByText(/USER NAME/)
        expect(text).toBeInTheDocument();
    });
})