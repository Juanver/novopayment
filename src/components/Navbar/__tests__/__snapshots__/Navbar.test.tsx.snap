// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Should render 1`] = `
<div>
  <div
    class="sc-bcXHqe fhPbsz"
  >
    <img
      alt="Tu Prutect logo"
    />
    <div
      class="sc-gswNZR dbKPCA"
    >
      <p>
        USER NAME
      </p>
      <img
        alt="Arrow down"
      />
    </div>
  </div>
</div>
`;
